src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java

```diff
@@ -403,19 +410,25 @@ public FormValidation doCheckServer(@QueryParameter String value, @QueryParamete
             if(!Jenkins.get().hasPermission(Jenkins.ADMINISTER))
                 return FormValidation.ok();
 
+            Context ctx = null;
             try {
-                Hashtable<String,String> props = new Hashtable<String,String>();
-                if(managerDN!=null && managerDN.trim().length() > 0  && !"undefined".equals(managerDN)) {
+                Hashtable<String,Object> props = new Hashtable<>();
+                if(StringUtils.isNotBlank(managerDN)  && !"undefined".equals(managerDN)) {
                     props.put(Context.SECURITY_PRINCIPAL,managerDN);
                 }
-                if(managerPassword!=null && managerPassword.trim().length() > 0 && !"undefined".equals(managerPassword)) {
+                if(StringUtils.isNotBlank(managerPassword) && !"undefined".equals(managerPassword)) {
                     props.put(Context.SECURITY_CREDENTIALS,managerPassword);
                 }
+                // normal
                 props.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
                 props.put(Context.PROVIDER_URL, LDAPSecurityRealm.toProviderUrl(server,rootDN));
 
-                DirContext ctx = new InitialDirContext(props);
-                ctx.getAttributes("");
+                props.put("java.naming.referral", "follow");
+                props.put("com.sun.jndi.ldap.connect.timeout", Integer.toString(CONNECT_TIMEOUT));
+                props.put("com.sun.jndi.ldap.connect.pool", "true");
```

Comment:
Pooling a one of check makes no sense

Reply:
answer here https://github.com/jenkinsci/ldap-plugin/pull/102#discussion_r705056385